types/wlr_output.c

```diff
@@ -454,6 +453,10 @@ void wlr_output_cursor_set_surface(struct wlr_output_cursor *cursor,
 		wl_signal_add(&surface->events.destroy, &cursor->surface_destroy);
 		output_cursor_commit(cursor);
 	} else {
+		cursor->enabled = false;
+		cursor->width = 0;
+		cursor->height = 0;
```

Comment:
Why are you resetting the width/height? Are these things ever read if the cursor is not enabled?

Reply:
Yeah, that's doesn't matter a lot. It's just to avoid having old state data.